core/src/main/java/org/springframework/security/authorization/AuthorizationManagers.java

```diff
@@ -56,6 +56,43 @@ public static <T> AuthorizationManager<T> anyOf(AuthorizationManager<T>... manag
 		};
 	}
 
+	/**
+	 * Creates an {@link AuthorizationManager} that grants access if at least one
+	 * {@link AuthorizationManager} granted, if <code>managers</code> are empty then
+	 * denied decision is returned. When all {@link AuthorizationManager}s abstained, a
+	 * default {@link AuthorizationDecision} is returned.
+	 * @param <T> the type of object that is being authorized
+	 * @param allAbstainDefaultDecision the default decision if all
+	 * {@link AuthorizationManager}s abstained
+	 * @param managers the {@link AuthorizationManager}s to use
+	 * @return the {@link AuthorizationManager} to use
+	 * @since 6.3
+	 */
+	@SafeVarargs
+	public static <T> AuthorizationManager<T> anyOf(AuthorizationDecision allAbstainDefaultDecision,
+			AuthorizationManager<T>... managers) {
+		return (authentication, object) -> {
+			if (managers.length == 0) {
```

Comment:
I wasn’t sure how to handle the case with an empty list of managers, and decided to check how java streams process such cases.
In docs for anyMatch(), "If the stream is empty then false is returned and the predicate is not evaluated."
For allMatch(), "If the stream is empty, the quantification is said to be vacuously satisfied and is always true (regardless of P(x))" https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/stream/Stream.html#anyMatch(java.util.function.Predicate) 
They follow the concept of vacuous truth to handle such cases.

Other methods without allAbstainDefaultDecision also return false in anyOf() and true in allOf() if the list of managers is empty.
I thought it was the logic for such a behaviour, so I added a check for an empty list in new methods.

Reply:
Thanks for the explanation and for updating the behavior. I can see why you'd look to Java streams for insight.

I think the difference here is that these methods are about allowing you to change the default empty behavior. It would be somewhat the same as Java streams having a feature like:

```
anyMatch(boolean defaultResponseIfEmpty)
```

which overrides the default behavior of `true`. 

The intent is to align this with an existing `AccessDecisionVoter` feature, `allowIfAllAbstainDecisions` to allow for easier migration from voters to authorization managers.

Does that resolve your concern?

Reply:
Yes, thank you
I agree with your comment